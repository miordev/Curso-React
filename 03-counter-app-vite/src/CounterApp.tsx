import PropTypes from 'prop-types';
import { useState } from 'react';

export const CounterApp = ({ value }: any) => {
  const [counter, setCounter] = useState(value);

  const handleAdd = () => {
    setCounter(counter + 1);
    // setCounter((c) => c + 1);
  };

  const handleSubtract = () => {
    setCounter(counter - 1);
  };

  const handleReset = () => {
    setCounter(value);
  };

  return (
    <>
      <h1>CounterApp</h1>
      <h2 data-testid="test-counter">{counter}</h2>
      <button aria-label="increase-btn" onClick={handleAdd}>
        +1
      </button>
      <button aria-label="decrease-btn" onClick={handleSubtract}>
        -1
      </button>
      <button aria-label="reset-btn" onClick={handleReset}>
        Reset
      </button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};
