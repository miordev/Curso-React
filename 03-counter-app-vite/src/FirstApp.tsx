import PropTypes from 'prop-types';

export const FirstApp = ({ title, subtitle, name }: any) => {
  return (
    <>
      <h1 data-testid="test-title">{title}</h1>
      <p>{subtitle}</p>
      <p>{subtitle}</p>
      <p>{name}</p>
    </>
  );
};

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired,
};

FirstApp.defaultProps = {
  title: 'There is not title',
  subtitle: 'There is not subtitle',
  name: 'Miguel Ordo√±ez',
};
