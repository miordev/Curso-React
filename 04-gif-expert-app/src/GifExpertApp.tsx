import { useState } from 'react';
import { AddCategory, GifGrid } from './components';

export const GifExpertApp = () => {
  const [categoryList, setCategoryList] = useState(['One Punch']);

  const onAddCategory = (newCategory: string): void => {
    if (!categoryList.includes(newCategory)) {
      setCategoryList([newCategory, ...categoryList]);
    }
  };

  return (
    <>
      <h1>GifExpertApp 2</h1>

      <AddCategory onNewCategory={onAddCategory} />

      {categoryList.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
    </>
  );
};
