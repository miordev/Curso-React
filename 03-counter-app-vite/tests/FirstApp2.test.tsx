import { render, screen } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';

describe('Tests in <FirstApp />', () => {
  const title = 'I am Miguel';
  const subtitle = 'This is a subtitle';

  test('should do match with the snapshot', () => {
    const { container } = render(<FirstApp title={title} />);
    expect(container).toMatchSnapshot();
  });

  test('should show the title', () => {
    render(<FirstApp title={title} />);
    expect(screen.getByTestId('test-title').innerHTML).toBe(title);
  });

  test('should show the title in h1', () => {
    render(<FirstApp title={title} />);
    expect(screen.getByRole('heading', { level: 1 })).toBeTruthy();
  });

  test('should show the subtitle', () => {
    render(<FirstApp title={title} subtitle={subtitle} />);
    expect(screen.getAllByText(subtitle).length).toBe(2);
  });
});
