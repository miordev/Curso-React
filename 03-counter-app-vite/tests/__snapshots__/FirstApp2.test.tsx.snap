// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <FirstApp /> should do match with the snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    I am Miguel
  </h1>
  <p>
    There is not subtitle
  </p>
  <p>
    There is not subtitle
  </p>
  <p>
    Miguel Ordo√±ez
  </p>
</div>
`;
